golang1.17/bin/compile

```diff
@@ -86,20 +86,32 @@ def build(source_dir, target_dir):
     }
 
     gomod = "%s/go.mod" % source_dir
+    vendor = "%s/vendor" % source_dir
+
+    has_gomod = exists(gomod)
+    deps_vendored = exists(vendor)
+
+    if has_gomod:
+        print("detected go.mod file")
+    if deps_vendored:
+        print("Detected vendor directory. Will skip downloading and initialization modules.")
+
     if os.environ.get("__NIM_REMOTE_BUILD"):
-        if exists(gomod):
-            print("downloading modules")
-            ret = subprocess.call(["go", "mod", "download"], cwd=source_dir, env=env)
-            if ret != 0:
-                sys.exit(ret)
+        if has_gomod:
+            if not deps_vendored:
```

Comment:
Nested `if` looks weird. Open to suggestions on how to clean up this whole build script.

Reply:
Nested if is needed because of the way the build needs to work. It needs to skip the go mod init step the same way it did before.